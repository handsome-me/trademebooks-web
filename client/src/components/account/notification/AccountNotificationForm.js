import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import { MDBBtn } from 'mdbreact'
import { toastr } from 'react-redux-toastr'

const AccountNotificationForm = ({
  getAccountSettings,
  saveAccountSettings
}) => {
  const [formData, setFormData] = useState({
    receiveEmail: false,
    receiveSms: false
  })

  const { receiveEmail, receiveSms } = formData

  const onChange = (e) => {
    setFormData({ ...formData, [e.target.name]: e.target.checked })
  }

  const saveSettings = async (e) => {
    e.preventDefault()

    const response = await saveAccountSettings(formData)

    if (response) {
      toastr.success('Your settings have been updated.')
    }
  }

  useEffect(() => {
    ;(async () => {
      const account = await getAccountSettings()

      setFormData(account)
    })()
  }, [getAccountSettings])

  return (
    <>
      <h3 className="mb-4 font-weight-bold">Notifications Settings</h3>

      <div className="custom-control custom-checkbox my-3">
        <input
          type="checkbox"
          name="receiveEmail"
          className="custom-control-input"
          id="defaultInline1"
          checked={receiveEmail}
          onChange={onChange}
        />
        <label className="custom-control-label" htmlFor="defaultInline1">
          Receive Emails
        </label>
      </div>

      <div className="custom-control custom-checkbox my-3">
        <input
          type="checkbox"
          name="receiveSms"
          className="custom-control-input"
          id="defaultInline2"
          checked={receiveSms}
          onChange={onChange}
        />
        <label className="custom-control-label" htmlFor="defaultInline2">
          Receive Text Messages
        </label>
      </div>

      <div>
        <MDBBtn onClick={saveSettings}>Save Changes</MDBBtn>
      </div>
    </>
  )
}

AccountNotificationForm.propTypes = {
  getAccountSettings: PropTypes.func.isRequired,
  saveAccountSettings: PropTypes.func.isRequired
}

export default AccountNotificationForm
