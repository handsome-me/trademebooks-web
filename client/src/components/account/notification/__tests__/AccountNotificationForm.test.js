import React from 'react'
import renderer from 'react-test-renderer'

import AccountNotificationForm from '../AccountNotificationForm'

describe('Component: AccountNotificationForm', () => {
  const props = {
    getAccountSettings: jest.fn(),
    saveAccountSettings: jest.fn()
  }

  it('should match the snapshot', () => {
    const accountNotificationFormSnapshot = renderer.create(<AccountNotificationForm {...props} />).toJSON()
    expect(accountNotificationFormSnapshot).toMatchSnapshot()
  })
})
