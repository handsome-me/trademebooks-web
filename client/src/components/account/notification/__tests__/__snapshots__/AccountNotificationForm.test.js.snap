// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AccountNotificationForm should match the snapshot 1`] = `
Array [
  <h3
    className="mb-4 font-weight-bold"
  >
    Notifications Settings
  </h3>,
  <div
    className="custom-control custom-checkbox my-3"
  >
    <input
      checked={false}
      className="custom-control-input"
      id="defaultInline1"
      name="receiveEmail"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="custom-control-label"
      htmlFor="defaultInline1"
    >
      Receive Emails
    </label>
  </div>,
  <div
    className="custom-control custom-checkbox my-3"
  >
    <input
      checked={false}
      className="custom-control-input"
      id="defaultInline2"
      name="receiveSms"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="custom-control-label"
      htmlFor="defaultInline2"
    >
      Receive Text Messages
    </label>
  </div>,
  <div>
    <button
      className="btn-default btn Ripple-parent"
      data-test="button"
      onClick={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      Save Changes
      <div
        className="Ripple "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
  </div>,
]
`;
